### DISCUSSION TITLE
Is it possible to map unstructured data to structured data?

### DISCUSSION THREAD

**Discussion by jackdh** [riok/mapperly#1118] [Q&A]
Say I have a list of JSONFields:

```c#
public class JsonField
{
    public string FieldCode { get; set; }
    public string FieldValue { get; set; }
}
```
```json
[
    {"FieldCode": "Main ", "FieldCode": "This is an address"},
    {"FieldCode": "Date", "FieldCode": "2023-02-01"},
    {"FieldCode": "Number", "FieldCode": "5"}
]
```

And want to map that list to a structured class

```c#
public class Address {
    public string Main { get; set; }
    public Date DateTaken  { get; set; }
    public int Number { get; set; }
}
```

Is it possible to have mapperly auto map that?

**Comment by latonz** [comment]
Seems to be related to https://github.com/riok/mapperly/issues/1041 and https://github.com/riok/mapperly/discussions/1039, but here the property names are in the value of a given key-value pair. For now this seems to be out of scope of Mapperly.
